File: RotateImage/image.js
// We are given a matrix that represents image coords
// The goal is to rotate the image by 90 degrees
// In other words: flip column & reverse row

function rotateImage(matrix) {
    // Step 1: for each row in matrix, reverse its order
    matrix.forEach(row => row.reverse());
    // Step 2: set up the loops needed to destructure
    for (let i = 0; i < matrix.length; i++) {
        for (let j = 0; j < i; j++) {
            // Step 3: reverse and destructure each index
            [matrix[i][j], matrix[j][i]] = [matrix[j][i], matrix[i][j]];
        }
    }
    return matrix.forEach(m => console.log(m));
}

console.log(rotateImage([
    [1, 2],
    [3, 4]
  ]));

  
console.log(rotateImage([
    [1, 1, 5, 9, 9],
    [2, 2, 6, 0, 0],
    [3, 3, 7, 1, 1],
    [4, 4, 8, 2, 2],
    [5, 5, 9, 3, 3]
]));

File: RotateImage/image.py
def rotate_image(img):
    for row in img:
        row.reverse()
    for i in range(len(img)):
        for j in range(i):
            img[i][j], img[j][i] = img[j][i], img[i][j]
    return img

print(rotate_image([
  [1, 2],
  [3, 4]
]))  # should print [ [ 2, 4 ], [ 1, 3 ] ]

print(rotate_image([
  [1, 1, 5, 9, 9],
  [2, 2, 6, 0, 0],
  [3, 3, 7, 1, 1],
  [4, 4, 8, 2, 2],
  [5, 5, 9, 3, 3]
]))  # should print [ [ 9, 0, 1, 2, 3 ],
     #                [ 9, 0, 1, 2, 3 ],
     #                [ 5, 6, 7, 8, 9 ],
     #                [ 1, 2, 3, 4, 5 ],
     #                [ 1, 2, 3, 4, 5 ] ]